Sources/UI/Device/Device.swift

```diff
@@ -0,0 +1,405 @@
+//
+//  Device.swift
+//  RoverUI
+//
+//  Created by Andrew Clunis on 2018-11-20.
+//  Copyright © 2018 Rover Labs Inc. All rights reserved.
+//
+
+import Foundation
+import os
+import UIKit
+import UserNotifications
+
+class Device {
+    
+    let userDefaults: UserDefaults
+    let jsonDecoder: JSONDecoder
+    let jsonEncoder: JSONEncoder
+    
+    let adSupportInfoProvider: AdSupportInfoProvider?
+    let bluetoothInfoProvider: BluetoothInfoProvider?
+    let telephonyInfoProvider: TelephonyInfoProvider?
+    let locationInfoProvider: LocationInfoProvider?
+    
+    let reachability = Reachability(hostname: "google.com")!
+    let userNotificationCenter = UNUserNotificationCenter.current()
```

Comment:
For same reasons as unnecessary injections, my preference is to refer to `UNUserNotificationCenter.current()` in the function where you actually need to use it instead of creating a constant on the class. It adds indirection as the user looking at the function implementation only knows he's dealing with an instance property named `userNotificationCenter` and has to follow the source to realize what you've done. If you create the instance in-line where you are using it, it adds clarity because developers are very accustomed to seeing `UNUserNotificationCenter.current()` and immediately understand, "ok you're accessing the UNUserNotifiicationCenter` singleton which is standard practice".